File: Easy/affidamentokind/meta_affidamentokinddefaultview/meta_affidamentokinddefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_affidamentokinddefaultview
{
    public class Meta_affidamentokinddefaultview : Meta_easydata 
	{
        public Meta_affidamentokinddefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "affidamentokinddefaultview") {
				Name = "Tipologie di affidamento";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idaffidamentokind"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "title", "Tipologia", nPos++);
						DescribeAColumn(T, "affidamentokind_description", "Descrizione", nPos++);
						DescribeAColumn(T, "affidamentokind_sortcode", "Ordinamento", nPos++);
						DescribeAColumn(T, "affidamentokind_active", "Attivo", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "default": {
						return "affidamentokind_sortcode";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/affidamentokind/meta_affidamentokinddefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_affidamentokinddefaultview() {
        MetaData.apply(this, ["affidamentokinddefaultview"]);
        this.name = 'meta_affidamentokinddefaultview';
    }

    meta_affidamentokinddefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_affidamentokinddefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'title', 'Tipologia', null, 20, 50);
						this.describeAColumn(table, 'affidamentokind_description', 'Descrizione', null, 30, 256);
						this.describeAColumn(table, 'affidamentokind_active', 'Attivo', null, 40, null);
						this.describeAColumn(table, 'affidamentokind_sortcode', 'Ordinamento', null, 50, null);
						this.describeAColumn(table, 'affidamentokind_costoora', 'Costo orario', 'fixed.2', 100, null);
						this.describeAColumn(table, 'affidamentokind_costoorariodacontratto', 'Costo orario da ruolo', null, 110, null);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idaffidamentokind"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "title asc ";
					}
					case "default": {
						return "title asc , affidamentokind_sortcode desc";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('affidamentokinddefaultview', new meta_affidamentokinddefaultview('affidamentokinddefaultview'));

	}());
